File: client/src/app/models/helpers/member-params.ts
import { PaginationParams } from "./paginationParams.model";

export class MemberParams extends PaginationParams {
  search: string = '';
  courseTitle: string = '';
  className: string = '';
  roles: string[] = ['student']
  minAge: number = 11;
  maxAge: number = 99;
}

File: backend/api/Helpers/MemberParams.cs
namespace api.Helpers;

public class MemberParams : PaginationParams
{   
    [MaxLength(100)]
    public string? Search { get; set; } = string.Empty;
    
    [Range(11, 99)]
    public int MinAge { get; set; }
    
    [Range(11, 99)]
    public int MaxAge { get; set; }
}